core/src/main/java/org/mobilitydata/gtfsvalidator/parsing/RowParser.java

```diff
@@ -216,6 +217,18 @@ public String asText(int columnIndex, boolean required) {
 
   @Nullable
   public String asId(int columnIndex, boolean required) {
+    String value = row.asString(columnIndex);
+    if (value == null) {
+      return asString(columnIndex, required);
```

Comment:
Why are you calling asString which is calling row.asString again if you already have the value?